plugin/PlayAssetDelivery/src/main/java/com/google/play/core/godot/assetpacks/utils/AssetPackStateFromDictionary.java

```diff
@@ -0,0 +1,97 @@
+/*
+ *  	Copyright 2020 Google LLC
+ *
+ *  	Licensed under the Apache License, Version 2.0 (the "License");
+ *  	you may not use this file except in compliance with the License.
+ *  	You may obtain a copy of the License at
+ *
+ *  		https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  	Unless required by applicable law or agreed to in writing, software
+ *  	distributed under the License is distributed on an "AS IS" BASIS,
+ *  	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  	See the License for the specific language governing permissions and
+ *  	limitations under the License.
+ */
+
+package com.google.play.core.godot.assetpacks.utils;
+
+import com.google.android.play.core.assetpacks.AssetPackState;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+import org.godotengine.godot.Dictionary;
+
+/**
+ * This class extends the AssetPackState abstract class, and provides constructor that allows
+ * PlayAssetDeliveryUtils class to instantiate AssetPackStateFromDictionary given a Godot
+ * Dictionary.
+ */
+public class AssetPackStateFromDictionary extends AssetPackState {
+  private String name;
+  private int status;
+  private int errorCode;
+  private long bytesDownloaded;
+  private long totalBytesToDownload;
+  private int transferProgressPercentage;
+
+  public static final String key1 = "name";
+  public static final String key2 = "status";
+  public static final String key3 = "errorCode";
+  public static final String key4 = "bytesDownloaded";
+  public static final String key5 = "totalBytesToDownload";
+  public static final String key6 = "transferProgressPercentage";
+
+  private static final Set<String> dictionaryRequiredKeySet =
+      new HashSet<>(Arrays.asList(key1, key2, key3, key4, key5, key6));
+
+  public AssetPackStateFromDictionary(Dictionary dict) throws IllegalArgumentException {
+    if (dict.keySet().containsAll(dictionaryRequiredKeySet)) {
+      try {
```

Comment:
I tried declaring public final Strings for the keys and referencing the declared Strings in the latest commit. In my opinion, this method can sacrifice some readability, especially for unfamiliar reader without IDEs, since the will have to figure out what is the actual value for key1.
If we name the Strings with more context such as AssetPackStateFromDictionary. transferProgressPercentage, I think it is more straightforward to just use "transferProgressPercentage".
I'm kind of not sure about this decision, what is your opinion?